src/components/element-editor/basic-details/elements/contact/default-country-code.tsx

```diff
@@ -0,0 +1,89 @@
+import { Alert } from "@lifesg/react-design-system/alert";
+import { Form } from "@lifesg/react-design-system/form";
+import { Controller, useFormContext } from "react-hook-form";
+import { TogglePair } from "src/components/common/toggle-pair/toggle-pair";
+import { IContactFieldAttributes } from "src/context-providers";
+import { MandatoryFieldBox } from "../../common";
+import { CountryData } from "./country-code";
+
+export const DefaultCountryCode = () => {
+    // ===========================================================================
+    // CONST, STATE, REFS
+    // ===========================================================================
+    const { control, watch, unregister } =
+        useFormContext<IContactFieldAttributes>();
+    const defaultCountryCode = watch("defaultCountryCode");
+    const displayAsFixedCountryCode = watch("displayAsFixedCountryCode");
+
+    // =========================================================================
+    // RENDER FUNCTIONS
+    // =========================================================================
+
+    return (
+        <MandatoryFieldBox>
+            <Controller
+                name="defaultCountryCode"
+                control={control}
+                render={({ field }) => {
+                    const { ref, ...fieldWithoutRef } = field;
+                    return (
+                        <Form.Select
+                            {...fieldWithoutRef}
+                            data-testid="default-country-code"
+                            label="Default country code (optional)"
+                            options={CountryData}
+                            selectedOption={CountryData.find(
+                                (item) => item[3] === field.value
+                            )}
+                            valueExtractor={(item) => item[3]}
+                            listExtractor={(item) => ({
+                                title: item[0] as string,
+                                secondaryLabel: `+${item[3] as string}`,
+                            })}
+                            displayValueExtractor={(item) =>
+                                `${item[0]} (+${item[3]})`
+                            }
+                            onSelectOption={(item) => field.onChange(item[3])}
```

Comment:
should clear the validation rule upon switching between Singapore and other country numbers

Reply:
Upon switching to other countries, validation rule would unmount and unregister its current values already. I have also set up a `useEffect` in `contact-validation-rule` (in the next branch) to set the value to the default Singapore validation upon first render. Does that logic sound ok?